--- qt-everywhere-opensource-src-4.8.1-clean/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h
+++ qt-everywhere-opensource-src-4.8.1-fixed/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h
@@ -52,7 +52,7 @@
 typedef struct _GHashTable GHashTable;
 typedef struct _GInputStream GInputStream;
 typedef struct _GList GList;
-typedef struct _GMutex GMutex;
+typedef union _GMutex GMutex;
 typedef struct _GPatternSpec GPatternSpec;
 typedef struct _GPollableOutputStream GPollableOutputStream;
 typedef struct _GSocketClient GSocketClient;
